src/Draco.Compiler/Api/CodeCompletion/SemanticCompletionProvider.cs

```diff
@@ -0,0 +1,77 @@
+using System.Collections.Immutable;
+using System.Linq;
+using Draco.Compiler.Api.Semantics;
+using Draco.Compiler.Api.Syntax;
+
+namespace Draco.Compiler.Api.CodeCompletion;
+
+/// <summary>
+/// Provides semantic completion.
+/// </summary>
+public sealed class SemanticCompletionProvider : CompletionProvider
```

Comment:
If this provides _all_ of semantic completion, then this provider is doing too much

Reply:
If it's only providing completion for member access, then it's fine, but naming is wrong